### PR TITLE
[DOC] Update README for C++ samples

### PR DESCRIPTION
**Pull request by generalova-kate**
### Details:
 - Update README with API info
 - Add docstrings to common functions

### Tickets:
 - 49428

### CODE REVIEW

File: inference-engine/samples/classification_sample_async/main.cpp [hunk 1]
```diff
@@ -60,46 +63,47 @@ int main(int argc, char *argv[]) {
...
         // -----------------------------------------------------------------------------------------------------
 
-        // --------------------------- 1. Load inference engine -------------------------------------
+        // --------------------------- Step 1. Initialize inference engine core -------------------------------------
         slog::info << "Creating Inference Engine" << slog::endl;
-
         Core ie;
+        // ------------------------------ Get Available Devices ------------------------------------------------------
+        slog::info << "Device info: " << slog::endl;
+        std::cout << ie.GetVersions(FLAGS_d);
```
Comment (mryzhov) [inline inference-engine/samples/classification_sample_async/main.cpp:82]: Don't we need to add here std::endl?

File: inference-engine/samples/classification_sample_async/main.cpp [hunk 2]
```diff
@@ -60,46 +63,47 @@ int main(int argc, char *argv[]) {
...
         }
-
-        /** Printing device version **/
-        std::cout << ie.GetVersions(FLAGS_d) << std::endl;
         // -----------------------------------------------------------------------------------------------------
 
-        // 2. Read a model in OpenVINO Intermediate Representation (.xml and .bin files) or ONNX (.onnx file) format
-        slog::info << "Loading network files" << slog::endl;
+        // Step 2. Read a model in OpenVINO Intermediate Representation (.xml and .bin files) or ONNX (.onnx file) format
+        slog::info << "Loading network files:"
```
Comment (mryzhov) [inline inference-engine/samples/classification_sample_async/main.cpp:98]:
```suggestion
            slog::info << "Loading network files:" << slog::endl << FLAGS_m << slog::endl;
```

File: inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/main.c [hunk 3]
```diff
@@ -167,7 +167,7 @@ void parseInputFilesArguments(int argc, char **argv) {
         readInputFilesArgument(argv[i]);
     }
 
-    if (file_num) {
+    if (file_num < 20) {
```
Comment (mryzhov) [inline inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/main.c:170]: What does this number mean? Maybe: int max_files = 20; if (file_num < max_files); ?

File: inference-engine/samples/classification_sample_async/main.cpp [hunk 4]
```diff
@@ -227,7 +234,7 @@ int main(int argc, char *argv[]) {
         const size_t resultsCnt = outputBlob->size() / batchSize;
         if (FLAGS_nt > resultsCnt || FLAGS_nt < 1) {
             slog::warn << "-nt " << FLAGS_nt << " is not available for this network (-nt should be less than " \
-                      << resultsCnt+1 << " and more than 0)\n            will be used maximal value : " << resultsCnt << slog::endl;
+                      << resultsCnt+1 << " and more than 0)\n            Maximal value " << resultsCnt << " will be used.";
```
Comment (mryzhov) [inline inference-engine/samples/classification_sample_async/main.cpp:237]: missed slog::endl;?

File: inference-engine/samples/common/utils/src/args_helper.cpp [hunk 5]
```diff
@@ -70,6 +72,15 @@ void parseInputFilesArguments(std::vector<std::string> &files) {
     for (auto img = img_begin; img != img_end; ++img) {
         readInputFilesArguments(files, *img);
     }
+
+    if (files.size() < 20) {
```
Comment (mryzhov) [inline inference-engine/samples/common/utils/src/args_helper.cpp:76]: one more number, looks like it can be a global constant or #define MAX_FILES 20?

**Review by mryzhov — CHANGES_REQUESTED** [review]
General comment, it is better to isolate all functional changes to separate PR, they can affect CI pipeline.

**Review by antonrom23 — APPROVED** [review]

**Review by mryzhov — APPROVED** [review]

**Pull request merged**